lib/fpm/package/deb.rb

```diff
@@ -641,29 +641,39 @@ def write_scripts
   end # def write_scripts
 
   def write_conffiles
-    return unless config_files.any?
-
-    # scan all conf file paths for files and add them
     allconfigs = []
-    config_files.each do |path|
+
+    # expand recursively a given path to be put in allconfigs
+    def add_path(path, allconfigs)
       # Strip leading /
       path = path[1..-1] if path[0,1] == "/"
       cfg_path = File.expand_path(path, staging_path)
+      Find.find(cfg_path) do |p|
+        allconfigs << p.gsub("#{staging_path}/", '') if File.file? p
```

Comment:
Since `Find.find` returns an `Enumerator`, I think moving the conditional `File.file?` to a `select` would make it easier to read:

```ruby
Find.find(cfg_path).select { |path| File.file?(path) }.each do |path|
  allconfigs << path.gsub("#{staging_path}/", "")
end
```